components/kyma-environment-broker/internal/ias/client_test.go

```diff
@@ -0,0 +1,313 @@
+)
+
+const (
+	companyID         = "sandbox"
+	serviceProviderID = "5e8c5e9e5f25d83ebec7e89e"
+	clientID          = "42"
+	clientSecret      = "floda"
+)
+
+func TestClient_GetCompany(t *testing.T) {
+	// given
+	server := fixHTTPServer(t)
+	defer server.Close()
+
+	client := NewClient(server.Client(), ClientConfig{URL: server.URL, ID: "admin", Secret: "admin123"})
+
+	// when
+	company, err := client.GetCompany()
+
+	// then
+	assert.NoError(t, err)
+	assert.Len(t, company.ServiceProviders, 1)
+	assert.Len(t, company.IdentityProviders, 1)
+}
+
+func TestClient_CreateServiceProvider(t *testing.T) {
+	// given
+	server := fixHTTPServer(t)
+	defer server.Close()
+
+	client := NewClient(server.Client(), ClientConfig{URL: server.URL, ID: "admin", Secret: "admin123"})
+
+	// when
+	err := client.CreateServiceProvider("someName", companyID)
+
+	// then
+	assert.NoError(t, err)
+}
+
+func TestClient_SetType(t *testing.T) {
+	// given
+	server := fixHTTPServer(t)
+	defer server.Close()
+
+	client := NewClient(server.Client(), ClientConfig{URL: server.URL, ID: "admin", Secret: "admin123"})
+
+	// when
+	iasType := Type{
+		SsoType:             "openID",
+		ServiceProviderName: "example.com",
+		OpenIDConnectConfig: OpenIDConnectConfig{
+			RedirectURIs:           []string{"https://example.com"},
+			PostLogoutRedirectURIs: nil,
+		},
+	}
+	err := client.SetType(serviceProviderID, iasType)
+
+	// then
+	assert.NoError(t, err)
+}
+
+func TestClient_SetAssertionAttribute(t *testing.T) {
+	// given
+	server := fixHTTPServer(t)
+	defer server.Close()
+
+	client := NewClient(server.Client(), ClientConfig{URL: server.URL, ID: "admin", Secret: "admin123"})
+
+	// when
+	attributes := PostAssertionAttributes{
+		AssertionAttributes: []AssertionAttribute{
+			{
+				AssertionAttribute: "first_name",
+				UserAttribute:      "firstName",
+			},
+		},
+	}
+	err := client.SetAssertionAttribute(serviceProviderID, attributes)
+
+	// then
+	assert.NoError(t, err)
+}
+
+func TestClient_SetSubjectNameIdentifier(t *testing.T) {
+	// given
+	server := fixHTTPServer(t)
+	defer server.Close()
+
+	client := NewClient(server.Client(), ClientConfig{URL: server.URL, ID: "admin", Secret: "admin123"})
+
+	// when
+	sni := SubjectNameIdentifier{
+		NameIDAttribute: "email",
+	}
+	err := client.SetSubjectNameIdentifier(serviceProviderID, sni)
+
+	// then
+	assert.NoError(t, err)
+}
+
+func TestClient_SetAuthenticationAndAccess(t *testing.T) {
+	// given
+	server := fixHTTPServer(t)
+	defer server.Close()
+
+	client := NewClient(server.Client(), ClientConfig{URL: server.URL, ID: "admin", Secret: "admin123"})
+
+	// when
+	auth := AuthenticationAndAccess{
+		ServiceProviderAccess: ServiceProviderAccess{
+			RBAConfig: RBAConfig{
+				RBARules: []RBARules{
+					{
+						Action:    "Allow",
+						Group:     "admins",
+						GroupType: "cloud",
+					},
+				},
+				DefaultAction: "Allow",
+			},
+		},
+	}
+	err := client.SetAuthenticationAndAccess(serviceProviderID, auth)
```

Comment:
with `SetFoo` method such test does not make sense because we do not assert if sth was really changed. Could you rewrite those test so it will not pass if someone will really change the contract?